### PR TITLE
fix: improve promtool health check documentation for authentication (#18807)

### LINKED ISSUE #18807: Promtool Healthcheck not working when basic authentication enabled
### What did you do?

I had a complete Prometheus workflow perfectly working with healthchecks enabled and so on.
I wanted to secure it a bit more by adding some basic authentication since some data can be "sensitive".

As soon as the basic authentication was added, the healthchecks stopped working.

The included `/bin/promtool check healthy` healthcheck in the docker image seems to not be working when basic authentication is enabled for prometheus.

### What did you expect to see?

It would be wise to disable basic auth for the health endpoints (such as /healthy or /ready) or adding the possibility to pass a flag "--web.config.file" so that promtool can use the same settings as prometheus.

### What did you see instead? Under which circumstances?

No real solution currently, I just disabled temporarily my healthchecks waiting for a solution.

### System information

Linux arch x86_64

### Prometheus version

```text
prometheus, version 3.11.3 (branch: HEAD, revision: eb173f5256d4022afba1e9bc3d19740a76859fae)
  build user:       root@83aad33dd38e
  build date:       20260427-14:45:32
  go version:       go1.26.2
  platform:         linux/amd64
  tags:             netgo,builtinassets
```

### Prometheus configuration file

```yaml
tls_server_config:
  cert_file: <redacted>
  key_file: <redacted>
basic_auth_users:
  <redacted>: <redacted>
```

### Alertmanager version

```text

```

### Alertmanager configuration file

```yaml

```

### Logs

```text
[user@svr statistic]# docker exec -it prometheus /bin/promtool check healthy --http.config.file=/etc/prometheus/tool.yml --url=https://localhost:9090

check failed: URL=https://localhost:9090, status=401
```

### PR DESCRIPTION
**Pull request by vigneshakaviki**
## Summary
Improves promtool health check documentation to clarify authentication support and adds comprehensive test cases.

### Documentation & Help Text
- Enhanced `httpConfigFileDescription` to explicitly mention authentication support
- Updated `healthy` command help to mention `--http.config.file` flag for authentication
- Updated `ready` command help to mention `--http.config.file` flag for authentication
- Added comprehensive documentation with examples for:
  - Basic authentication
  - TLS and basic authentication combined
  - Docker usage with health checks
  - Security considerations
  - Troubleshooting guide

### Test Coverage
Added test cases verifying:
- Health checks work correctly with basic auth credentials via HTTP config file
- Health checks properly fail (401) when credentials are missing

```release-notes
[ENHANCEMENT] Improves promtool health check documentation to clarify that --http.config.file flag supports authentication (basic auth, TLS, OAuth2). Adds test cases verifying basic auth functionality works correctly.
```

## Fixes
Fixes #18807

### CODE REVIEW

**Comment by krajorama** [comment]
Hello from the bug scrub !

Please fix the CI failures, otherwise this looks legit.

File: documentation/examples/promtool-healthcheck-with-auth.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
+# Using Promtool Health Check with Authentication
```
Comment (krajorama) [inline documentation/examples/promtool-healthcheck-with-auth.md:1]: The examples directory only really has config examples, I think this should go under https://github.com/prometheus/docs/tree/main/docs/guides in the docs repo, to appear in https://prometheus.io/docs/guides

File: docs/command-line/promtool.md [hunk 2]
```diff
@@ -133,31 +133,31 @@ Check if the web config files are valid or not.
...
+| <code class="text-nowrap">--http.config.file</code> | HTTP client configuration file (supports TLS, basic auth, OAuth2, etc.), see details at https://prometheus.io/docs/prometheus/latest/configuration/promtool |  |
 | <code class="text-nowrap">--url</code> | The URL for the Prometheus server. | `http://localhost:9090` |
 
 
 
 
 ##### `promtool check ready`
 
-Check if the Prometheus server is ready.
+Check if the Prometheus server is ready. Use `--http`.config.file flag to pass authentication credentials.
```
Comment (krajorama) [inline docs/command-line/promtool.md:152]: This is rendered wrong by documentcli, the apostrophes are wrong. Let's change the wording to not include the argument name, something like "Authentication credentials can be supplied via the HTTP client config file.".

File: documentation/examples/promtool-healthcheck-with-auth.md [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+1. Verify credentials in your config file match those configured in Prometheus
+2. Check the HTTP config file path is correct and readable
+3. Ensure the `--url` matches the Prometheus server address
+
+Example debugging:
+
+```bash
+# Enable verbose output to see what's happening
+promtool check healthy --http.config.file=promtool-auth.yml --url=https://prometheus.example.com:9090 -v
```
Comment (krajorama) [inline documentation/examples/promtool-healthcheck-with-auth.md:90]: Is there really a `-v` flag?

File: cmd/promtool/main_healthcheck_test.go [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+	"os"
+	"path/filepath"
+	"testing"
+
+	promconfig "github.com/prometheus/common/config"
+	"github.com/stretchr/testify/require"
+)
+
+// TestCheckServerStatusWithBasicAuth verifies health check works with basic auth.
+func TestCheckServerStatusWithBasicAuth(t *testing.T) {
```
Comment (krajorama) [inline cmd/promtool/main_healthcheck_test.go:29]: The two tests are very similar could you merge them into one data/table driven test?

File: cmd/promtool/main_healthcheck_test.go [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+	// Create a test HTTP server that requires basic auth
```
Comment (krajorama) [inline cmd/promtool/main_healthcheck_test.go:30]: I think we can get rid of these docstrings, it pretty obvious from the code itself what's going on.

**Review by krajorama — COMMENTED** [review]
Thanks for this, I've added a couple of comments. In addition: in general we don't add release notes for doc, test changes, so I think release-notes should be NONE.
